contracts/cw721-base/src/contract.rs

```diff
@@ -478,12 +484,14 @@ fn query_all_approvals<S: Storage, A: Api, Q: Querier>(
     start_after: Option<HumanAddr>,
     limit: Option<u32>,
 ) -> StdResult<ApprovedForAllResponse> {
-    let owner_raw = deps.api.canonical_address(&owner)?;
     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
-    let start = calc_range_start_human(deps.api, start_after)?;
+    let start_canon = maybe_canonical(deps.api, start_after)?;
+    let start = OwnedBound::exclusive(start_canon);
```

Comment:
Nice, but is this really need? I think plainly doing
```suggestion
    let start_raw = deps.api.canonical_address(&start_after)?;
    let start = OwnedBound::exclusive(Some(start_raw));
```
is clearer.

Moreover, `exclusive()` can directly receive the value and build the option inside. There could be another helper for `OwnedBound::None`, if needed.

Reply:
The issue is `calc_range_start_human` was a "do everything" function. I could replace it, or just try to pull out the reusable parts.

`start_after` is `Option<HumanAddr>`, which is the complexity - doing `maybe_canonical` instead of `deps.api.canonical` (it just does a Option::map).  Your proposed line `let start_raw = deps.api.canonical_address(&start_after)?;` would not compile. However, I could inline the contents of `maybe_canonical`, which is just one line and maybe cleaner.

Reply:
I see the need for `maybe_canonical` now. I still don't like it, though :-).

I think the issue here is that you are still computing a `start_after`. Wouldn't be better to pass the original `HumanAddr` wrapped in an `Bound::Exclusive` container, and handle all the range stuff inside? Not sure if that collides with use cases, though.

Reply:
The json interface is "no field", or `start_after: "cosmos12345..."`, thus the option.

I don't want to expose the bound to json (which is where the `Option<HumanAddr>` comes from

Reply:
I see. Sorry about the previous comment, it was based on a misunderstanding. You are right, not exposing `Bound` to the user is a good idea.